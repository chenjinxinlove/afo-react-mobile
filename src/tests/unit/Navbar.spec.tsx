import {shallow} from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Button from '../../pageages/button/index';
import Navbar from '../../pageages/navbar/';
import DemoNavTag from '../../pageages/navbar/demo';

describe('Navbar', () => {
  // 测试是否能正确渲染
  it('props', () => {
    const wrapper = shallow(<Navbar />)
    expect(wrapper.hasClass('afo-navbar'))
  })
  // 测试props
  it('props', () => {
    const wrapper = shallow(<Navbar title="title" fixed={true} leftContent={<Button icon="back">返回</Button>} rightContent={<Button icon="more" />}/>)
    expect(wrapper.hasClass('is-fixed')).toEqual(true)
  })
  // 测试快照
  it('test snapshots', () => {
    const component = renderer.create(
      <DemoNavTag />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
