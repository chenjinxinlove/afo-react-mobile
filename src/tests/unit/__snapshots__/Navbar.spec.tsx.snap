// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test snapshots 1`] = `
<div>
  <header
    className="afo-navbar is-fixed"
  >
    <div
      className="afo-navbar-button is-left"
    >
      <button
        className="afo afo-button afo-button--default afo-button--normal"
        disabled={false}
        laoding={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="afo-icon-back"
        />
        返回
      </button>
    </div>
    <h1
      className="afo-navbar-title"
    >
      我是fixed标题
    </h1>
    <div
      className="afo-navbar-button is-right"
    >
      <button
        className="afo afo-button afo-button--default afo-button--normal"
        disabled={false}
        laoding={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="afo-icon-more"
        />
      </button>
    </div>
  </header>
</div>
`;
